test/core/integration/concrete/lockup/batch/batch.t.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.22 <0.9.0;
+
+import { Batch } from "src/core/abstracts/Batch.sol";
+import { Errors } from "src/core/libraries/Errors.sol";
+
+import { Integration_Test } from "./../../../Integration.t.sol";
+
+abstract contract Batch_Integration_Concrete_Test is Integration_Test {
+    function test_RevertWhen_CallFunctionNotExist() external {
+        bytes[] memory calls = new bytes[](1);
+        calls[0] = abi.encodeCall(dai.balanceOf, (users.recipient));
+
+        Batch(address(lockup)).batch(calls);
```

Comment:
Missting `expectRevert` statement.

Reply:
hmm, curious why the test isn't failing then, i guess the branch should say `it should do nothing`

Reply:
Did you figure out why its not failing?

Reply:
It did not revert because `dai.balanceOf` encodes to `balanceOf(address)` selector which is a valid function in `ERC721` contract.  I fixed it in https://github.com/sablier-labs/v2-core/pull/1070/commits/4865b3c15a340f314afdf53396c3d5f9cc51e55a.

-----------

I also want to mention that because the test was not failing, changing branch to `it should do nothing should not be a solution. This is an example of white box testing where you tried to curve fit the test to made them pass. Referring to my discussion on [white box testing vs black box testing](https://github.com/sablier-labs/flow/discussions/311), in a black box testing, we wont change the test but rather investigate the internal logic in such cases.

I look forward to see your comments on my discussion.